app/services/badge_granter.rb

```diff
@@ -44,6 +44,7 @@ def self.mass_grant(badge, users)
 
   def grant
     return if @granted_by && !Guardian.new(@granted_by).can_grant_badges?(@user)
+    return unless @badge.present? && @badge.enabled?
     return unless @badge.enabled?
```

Comment:
Did you forget to delete this line?

We don't need to check for it twice.

Reply:
Oh yeah, oops! Fixed.